CONTRIBUTING.md

```diff
@@ -4,13 +4,27 @@ This is a [Jazzband](https://jazzband.co/) project. By contributing you agree
 to abide by the [Contributor Code of Conduct](https://jazzband.co/about/conduct)
 and follow the [guidelines](https://jazzband.co/about/guidelines).
 
+## Getting started
+
+Install tox:
+
+    pip install tox
+
+Create a virtual environment, with `pip-tools` in development mode and its test dependencies,
+using your current python version:
+
+    tox --devenv .venv -e py
+
+Activate the virtual environment:
+
+    source ./.venv/bin/activate
+
 ## Project Contribution Guidelines
 
 Here are a few additional or emphasized guidelines to follow when contributing to pip-tools:
 
-- Install pip-tools in development mode and its test dependencies with `pip install -e .[testing]`.
 - Check with `tox -e checkqa` to see your changes are not breaking the style conventions.
-- Always provide tests for your changes.
+- Always provide tests for your changes, use `tox -e coverage` to run the tests with coverage reports.
```

Comment:
Honestly, running `tox` will run all the test envs with coverage + linters. Not sure if mentioning just `coverage` is needed. Besides, codecov will report back to PRs if it's decreased, I think. Maybe this sort of detail belongs to a development guide rather than contribution instructions. I see such details as non-mandatory things that may help during the development process but should be used if the developer actually needs them. Same with use `tox -e py` to run on test env with the current Python, use `tox -e pyXY` to specify other Python, and so on — we could basically end up documenting how tox works but that would probably be too overwhelming.
Mentioning `tox -av` along the lines might be useful, OTOH.

Reply:
`tox -e checkqa` was already there when I started contributing to `pip-tools`, so it already looked like a mix of contribution and development guide to me. 
We could eventually regroup these instructions in a Development guide section of this file? I think making another file for this is too much. I personally like a README that tells me exactly what to do to get started quickly. Later I can go to the tool's website to understand it better (we could provide the links too).

Also, when I ran `tox -e coverage` and `tox -e checkqa`, it ran the checks and the tests only for my version of python. Do you suggest that we mention to run `tox` only, to run everything once?

By the way what I did for my first PR:
- Ensured the tests were passing for my python version, checked the code coverage reports about my new tests
- Run the code linters which fixed my files
- See in my PR that the CI succeeded for every python versions

Reply:
Maybe `tox -p all`?

Reply:
FYI `tox -p all` is really slow on my old macbook air:

```
✔ OK py39-piplatest-coverage in 9 minutes, 36.482 seconds
✔ OK py39-pipprevious-coverage in 10 minutes, 20.35 seconds
✔ OK py39-pip20.3-coverage in 10 minutes, 20.76 seconds
✔ OK py39-pipmain-coverage in 10 minutes, 40.551 seconds
```

So do you guys consider running `tox` with all test environments mandatory?
Also it seems to run only for `py39`, the current version I am using. Does that mean devs should install other python versions on their machine so `tox` can use them as well? If so, this is something we should mention I think? (it's my first time contributing to an open source project and I am not used to how `tox` works)

According to what you say, I suggest to change `Getting started` to `Development Guidelines`. 
Inside, we could suggest command lines to get things done correctly. But it should be kept simple, as @webknjaz said, we don't want to write a full doc about the tools and all possibilities.

Inside the contributing guidelines, we could have:

```
- Always provide tests for your changes.
- Check with `tox` that your changes are not breaking the style conventions.
```

And inside `Development Guidelines`, we could have what we wrote in `Getting started` and add a line or two about testing the branch, probably `$ tox -p all`.

What do you think?